### ISSUE TITLE
🐛 assertion failed: start.raw <= end.raw

### ISSUE THREAD

**Issue by IDKHowToDev** [biomejs/biome#9244]
### Environment information

<details>

when running `pnpm biome lint . --diagnostic-level=error` on svlelte kit project

Paste output here

```Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:    

Source Location: C:\actions-runner\biome\biome\crates\biome_text_size\src\range.rs:67:9     
Thread Name: main
Message: assertion failed: start.raw <= end.raw
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

</details>


### What happened?

1. i ran `pnpm biome lint . --diagnostic-level=error`
2. biome started running perefectly fine on my svelte kit project 
3. biome suddently encountered a bug that couldn't be fixed 

### Expected result

it should of kept linting and showing me errors in my code with no problems but it didn't happen 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
